internal/node/proxy.go

```diff
@@ -34,7 +33,7 @@ type ProxyBus struct {
 	resolver    TwinResolver
 }
 
-func NewProxyBus(endpoint string, twinID uint32, sub subi.SubstrateExt, signer substrate.Identity, verifyReply bool) (*ProxyBus, error) {
+func NewProxyBus(endpoint string, twinID uint32, sub *substrate.Substrate, signer substrate.Identity, verifyReply bool) (*ProxyBus, error) {
 	if len(endpoint) != 0 && endpoint[len(endpoint)-1] == '/' {
 		endpoint = endpoint[:len(endpoint)-1]
```

Comment:
you can use `strings.TrimSuffix("/")`for this and no need to do len check, the zero value of string is empty and totally valid to use in subsequent calls.